qiskit/opflow/primitive_ops/pauli_sum_op.py

```diff
@@ -202,7 +202,12 @@ def permute(self, permutation: List[int]) -> "PauliSumOp":
                 "List of indices to permute must have the same size as Pauli Operator"
             )
         length = max(permutation) + 1
-        spop = self.primitive.tensor(SparsePauliOp(Pauli("I" * (length - self.num_qubits))))
+
+        if length > self.num_qubits:
+            spop = self.primitive.tensor(SparsePauliOp(Pauli("I" * (length - self.num_qubits))))
+        else:
+            spop = self.primitive
```

Comment:
Pre-existing, but this will give incorrect results if the user passes a permutation with duplicates in.  Is it worth adding a `if len(set(permutation)) != len(permutation): raise` block in error checking at the top as well?  Also, you can currently pass negative numbers, and strange things will happen - you only get an error with sufficiently large negative numbers (`-(self.num_qubits - 1)`).  Should we fix them while we're at it?

Reply:
Good points. See here: 6481924